builder/tests/common-types/src/main/java/io/helidon/common/types/AnnotationSupport.java

```diff
@@ -619,22 +627,33 @@ private static Class<?> asClass(TypeName typeName, String property, Object value
         if (value instanceof Class<?> theClass) {
             return theClass;
         }
-        if (value instanceof String str) {
-            try {
-                return Class.forName(str);
-            } catch (ClassNotFoundException e) {
-                throw new IllegalArgumentException(typeName.fqName() + " property " + property
-                                                           + " of type String and value \"" + str + "\""
-                                                           + " cannot be converted to Class");
-            }
+
+        String className;
+
+        if (value instanceof TypeName tn) {
```

Comment:
(This block could be a simple `switch` expression)

Reply:
Fixed (and fixed another problem in this method).